ntbs-service/Pages/Notifications/Edit/ClinicalDetails.cshtml.cs

```diff
@@ -111,26 +111,31 @@ protected override async Task<bool> ValidateAndSave()
             }
             
             var notificationSites = CreateNotificationSitesFromModel(Notification);
+            
             ClinicalDetails.SetFullValidation(Notification.NotificationStatus);
             OtherSite?.SetFullValidation(Notification.NotificationStatus);
-            // Separate notification with notification sites only is needed to check if notification sites are valid,
-            // and to avoid updating notification site when updating Clinical Details
-            var notificationWithSitesOnly = new Notification {
-                ShouldValidateFull = Notification.ShouldValidateFull,
-                NotificationSites = notificationSites.ToList()
-            };
             
-            var isValid = TryValidateModel(this);
-            // Validate notification with sites regardless previous validation result
-            isValid = TryValidateModel(notificationWithSitesOnly, notificationWithSitesOnly.GetType().Name) && isValid;
+            // Since notification has other properties which are not populated by this page but have validation rules, 
+            // validation of a whole Notification model will result in validation errors.
+            // Therefore we need to manually validate NotificationSites and TryValidate other models
+            if (Notification.ShouldValidateFull && notificationSites.Count() == 0) 
+            {
+                ModelState.AddModelError("Notification.NotificationSites", ValidationMessages.DiseaseSiteIsRequired);
+            }
+            
+            TryValidateModel(ClinicalDetails, ClinicalDetails.GetType().Name);
+            if (OtherSite != null) 
+            {
+                TryValidateModel(OtherSite, OtherSite.GetType().Name);
+            }
 
-            if (isValid)
+            if (ModelState.IsValid)
             {
                 await service.UpdateClinicalDetailsAsync(Notification, ClinicalDetails);
                 await service.UpdateSitesAsync(Notification.NotificationId, notificationSites);
             }
 
-            return isValid;
+            return ModelState.IsValid;
```

Comment:
Now that we're not returing a custom variable, but just a property of the `ModelState`, to which the calling context has access to (as it's a method on the parent), should we stop returning it and make parent look at `ModelState` itself?

Reply:
I will need to go through other pages to make sure that it won't break anything